### ISSUE TITLE
Rule Change: Allow-list/-regex for no-underscore-dangle

### ISSUE THREAD

**Issue by besfahbod** [eslint/eslint#16334]
### What rule do you want to change?

no-underscore-dangle 

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
mutate({
  mutation:
    someMutation,
  variables: {
    // some variables...
    connections: [
      queryRoot.myConnection.__id,
    ],
  },
```

### What does the rule currently do for this code?

Raise warning/error on the `.__id` usage.

### What will the rule do after it's changed?

Won't raise warning/error on using `.__id`, when `__id` is explicitly marked as allowed when enabling the rule.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The idea is to enable projects to "allow" some specific use-cases of dangling underscores, specially for cases where the name is controlled by an external project/spec.

For example, recent versions of GraphQL/Relay clients use a new object property name called `__id`, which is expected to be used by application code. The dunder prefix in these names are indicators of the properties having a "special" status on the objects.

At the moment, there's no way to tell `no-underscore-dangle` rule to leave these special names alone, resulting in every use of them having to be explicitly (and manually) silenced, which is bad for developer experience.

A possible solution here would be to add a new rule option to pass in a list of names (or globs, or regex) of names allowed to have dangling underscores.

**Comment by kamilio** [comment]
This would be such a useful feature!

**Comment by mdjermanovic** [comment]
Hi @besfahbod, thanks for the issue!

This option already exists: https://eslint.org/docs/latest/rules/no-underscore-dangle#allow

Here's a [playground example](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tdW5kZXJzY29yZS1kYW5nbGVcIjogW1wiZXJyb3JcIiwgeyBcImFsbG93XCI6IFtcIl9faWRcIl0gfV0gKi9cblxubXV0YXRlKHtcbiAgbXV0YXRpb246XG4gICAgc29tZU11dGF0aW9uLFxuICB2YXJpYWJsZXM6IHtcbiAgICAvLyBzb21lIHZhcmlhYmxlcy4uLlxuICAgIGNvbm5lY3Rpb25zOiBbXG4gICAgICBxdWVyeVJvb3QubXlDb25uZWN0aW9uLl9faWQsIC8vIG5vIGVycm9yXG4gICAgXSxcbiAgfVxufSkiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==) with `"no-underscore-dangle": ["error", { "allow": ["__id"] }]`

**Comment by besfahbod** [comment]
Thanks, @mdjermanovic, for the quick response! Oops! Looks like I misunderstood the `allow` option from the docs. Great to see it's already available!
